packages/library/src/commonMain/kotlin/io/realm/RealmModel.kt

```diff
@@ -21,5 +21,5 @@ package io.realm
 //  https://github.com/realm/realm-kotlin/issues/83
 
 fun RealmObject.delete() {
-    Realm.delete(this)
+    MutableRealm.delete(this)
```

Comment:
Maybe remove this as it is leaking mutability outside the `write/writeBlocking`-scope. If the method was on `RealmObject` itself I guess you could scope by a `MutableRealm`-receiver, so it is only available on `RealmObject` inside the write function.

Reply:
Yes, we should probably just move this to an instance method on MutableRealm 👍

Reply:
Also created https://github.com/realm/realm-kotlin/issues/181